### DISCUSSION TITLE
Return a base64 string after hooking into the middleware

### DISCUSSION THREAD

**Discussion by rndthoughts** [SixLabors/ImageSharp.Web#180] [Q&A] [answered]
I am using the Azure Blob and Cache providers and would like to know if there is a way I can hook into the middleware to retrieve an image and hit the cache if applicable so that I can then take this image and return it as base64?

I looked at implementing custom providers and I can see how to do it but obviously any provider implementing `ImageWebProcessor` needs to return a `FormattedImage` which presumably gets passed to the caller usually via the url request.

I need to return a base64 string, but would like to make use of any existing cached image objects to avoid the extra processing?

**Comment by JimBobSquarePants** [comment, marked as answer]
I’m afraid it’s simply not designed for that. 

Tbh I’m not sure what the possible use case would be either?

**Reply by rndthoughts** [reply]
I have code that generates a pdf into which I would like to inject the profile image of a user.  It would be great to be able to call the image middleware with a given size, let it cache it and then hit the cache next time before converting to base64.  Otherwise I have to resize everytime.

**Reply by JimBobSquarePants** [reply]
Does the size change often? Why not resize it once on upload and store it.
